lms/djangoapps/affiliates/decorators.py

```diff
@@ -0,0 +1,44 @@
+from django.http import HttpResponseNotFound
+from functools import wraps
+from .models import AffiliateMembership
+
+def only_program_director(function):
+    """Only allow access to global staff or affiliate staff user(Program Director)"""
+    @wraps(function)
+    def wrapped_view(request, *args, **kwargs):
+        """Wrapper for the view function."""
+        if request.user.is_anonymous():
+            return HttpResponseNotFound()
+        elif request.user.is_staff:
+            return function(request, *args, **kwargs)
+        else:
+            has_pd_role_in_affiliate = AffiliateMembership.objects.filter(member=request.user, affiliate_id=kwargs['slug'], role='staff').exists()
+
+            if has_pd_role_in_affiliate:
+                return function(request, *args, **kwargs)
+            else:
+                return HttpResponseNotFound()
+
+    return wrapped_view
+
+
+def only_staff(function):
+    """Only allow access to global staff or affiliate staff user(Program Director)"""
+    @wraps(function)
+    def wrapped_view(request, *args, **kwargs):
+        """Wrapper for the view function."""
+        if request.user.is_anonymous():
+            return HttpResponseNotFound()
+        elif request.user.is_staff:
+            return function(request, *args, **kwargs)
+        else:
+            has_pd_role_in_affiliate = AffiliateMembership.objects.filter(member=request.user, affiliate_id=kwargs['slug'], role='staff').exists()
```

Comment:
DRY this up. We don't need 2 queries for each role. We can look if role is in array.